docs/yul.rst

```diff
@@ -174,9 +174,37 @@ whitespace, i.e. there is no terminating ``;`` or newline required.
 Literals
 --------
 
-As literals, you can use integer constants in decimal or hexadecimal notation
```

Comment:
I think this was kept generic, ignoring the dialect, but in practice, we enforced aspects of the EVM dialect more and more over time, so that there is currently no yul dialect that allows words larger than 32 bytes.

The actual semantics highly depend on the dialect and this mostly includes the interpretation of literals and bulit-in functions. The built-in functions are not explained in detail, so maybe we should also not go that much in detail here?

On the other hand, we could also add some more details in the built-in functions part like explain how the builtins that actually interpret the data (add, sub, ...) instead of just moving data around.

Further down it says "When compiling for the EVM..." - so maybe we should combine that part with the part here that explain how the syntax of literals are interpreted and just explain the syntax in a dialect-free first part?